crates/memory/src/op_store.rs

```diff
@@ -53,33 +56,59 @@ impl OpStore for Kitsune2MemoryOpStore {
         .boxed()
     }
 
+    /// Store the combined hash of a time slice.
+    ///
+    /// The `slice_id` is the index of the time slice. This is a 0-based index. So for a given
+    /// time period being used to slice time, the first `slice_hash` at `slice_id` 0 would
+    /// represent the combined hash of all known ops in the time slice `[0, period)`. Then `slice_id`
+    /// 1 would represent the combined hash of all known ops in the time slice `[period, 2*period)`.
     fn store_slice_hash(
         &self,
         slice_id: u64,
-        slice_hash: Vec<u8>,
+        slice_hash: bytes::Bytes,
     ) -> BoxFuture<'_, K2Result<()>> {
         async move {
             self.write()
                 .await
                 .time_slice_hashes
-                .insert(slice_id, slice_hash);
-            Ok(())
+                .insert(slice_id, slice_hash)
         }
         .boxed()
     }
 
+    /// Retrieve the count of time slice hashes stored.
+    ///
+    /// Note that this is not the number of hashes that have been provided at unique `slice_id`s.
+    /// Start from time slice id 0 and count up to the highest stored id. This value should be the
+    /// count based on the highest stored id.
+    /// This value is easier to compare between peers because it tracks an absolute number of time
+    /// slices that have had a slice hash computed and stored. The number of stored hashes would
```

Comment:
Hmm, yeah this isn't clear and looking at it again, it's not entirely true... Because we'll sync from time 0 but also be doing "latest ops" from peers and we'll end up computing slices at both ends potentially during a catch up...

Let me have another think about this

Reply:
I've had a go at justifying what I was thinking here. Please take a look at the updated documentation and if it doesn't make more sense then please make me try again, because it might just not make sense :)